core/src/server/mod.rs

```diff
@@ -42,3 +42,65 @@ pub use helpers::{BatchResponseBuilder, BoundedWriter, MethodResponse, MethodSin
 pub use host_filtering::*;
 pub use rpc_module::*;
 pub use subscription::*;
+
+use jsonrpsee_types::{ErrorObjectOwned, ResponsePayload};
+
+/// Something that can be converted into a JSON-RPC method call response.
+///
+/// If the value couldn't be serialized/encoded, jsonrpsee will sent out an error
+/// to the client `response could not be serialized`.
+pub trait IntoResponse {
+	/// Output.
+	type Output: serde::Serialize + ToOwned<Owned = Self::Output>;
+
+	/// Something that can be converted into a JSON-RPC method call response.
+	fn into_response(self) -> ResponsePayload<'static, Self::Output>;
+}
+
+impl<T, E: Into<ErrorObjectOwned>> IntoResponse for Result<T, E>
+where
+	T: serde::Serialize + ToOwned<Owned = T>,
+{
+	type Output = T;
+
+	fn into_response(self) -> ResponsePayload<'static, Self::Output> {
+		match self {
+			Ok(val) => ResponsePayload::result(val),
+			Err(e) => ResponsePayload::Error(e.into()),
+		}
+	}
+}
+
+macro_rules! impl_into_response {
+	($($n:ty),*) => {
+		$(
+			impl IntoResponse for $n {
+				type Output = $n;
+
+				fn into_response(self) -> ResponsePayload<'static, Self::Output> {
+					ResponsePayload::result(self)
+				}
+			}
+		)+
+	}
+}
+
+impl_into_response!(
```

Comment:
we could remove these because it doesn't work to implement these for `[T], Vec<T>` and other sequence types because of the result impl above